src/tests/index.test.js

```diff
@@ -1,16 +1,476 @@
 'use strict';
 
 var expect = require('expect.js'),
-    Plugin = require('../index.js');
+    Plugin = require('../index.js'),
+    BbPromise = require('bluebird'),
+    sinon = require('sinon');
 
 describe('serverless-plugin-external-sns-events', function() {
 
+   function createMockServerless(requestFunc) {
+
+      var serverless, provider;
+
+      provider = {
+         request: requestFunc
+      };
+
+      serverless = {
+         getProvider: function(providerName) {
+
+            if (providerName !== 'aws') {
+               return null;
+            }
+
+            return provider;
+         },
+         service: {
+            provider: {
+               compiledCloudFormationTemplate: {
+                  Resources: {}
+               }
+            }
+         },
+         cli: { log: function() {
+            return;
+         } }
+         // cli: { log: function(val) {
+         //    process.stdout.write(val + '\n');
+         // } }
+      };
+
+      return serverless;
+
+   }
+
+   function createMockRequest(requestStub) {
+
+      return function() {
+
+         var reqArgs = Array.prototype.slice.call(arguments);
+
+
+         return new BbPromise(function(resolve, reject) {
+            var result;
+
+            result = requestStub.apply(undefined, reqArgs);
+            if (result !== null) {
+               resolve(result);
+               return;
+            }
+
+            reject(new Error('Call to request() with unexpected arguments:  ' +
+            JSON.stringify(reqArgs)));
```

Comment:
Indentation doesn't match standards. I'm again confused why eslint didn't catch this. Please reformat.